src/libsyntax_ext/format.rs

```diff
@@ -626,7 +753,9 @@ pub fn expand_preparsed_format_args(ecx: &mut ExtCtxt, sp: Span,
                                     name_ordering: Vec<String>,
                                     names: HashMap<String, P<ast::Expr>>)
                                     -> P<ast::Expr> {
-    let arg_types: Vec<_> = (0..args.len()).map(|_| None).collect();
+    // We can't use vec! here for Vec<ArgumentType>, as ArgumentType isn't Clone.
+    let arg_types: Vec<_> = (0..args.len()).map(|_| Vec::new()).collect();
+    let arg_unique_types: Vec<_> = (0..args.len()).map(|_| Vec::new()).collect();
```

Comment:
Nope, `ArgumentType` is not `Clone`, so `vec![something; num]`, which is `vec::from_elem` under the hood, will fail on this.

```rust
#[derive(PartialEq)]
enum ArgumentType {
    Placeholder(String),
    Count,
}

fn main() {
    let a: Vec<Vec<ArgumentType>> = vec![Vec::new(); 100];
}
```

No matter how you say `Vec::new()` this won't compile. Either we make `ArgumentType` `Clone` which I assume is not really desirable, or keep using the current way, which isn't that ugly after all.